xapian-core/include/xapian/weight.h

```diff
@@ -526,15 +526,20 @@ class XAPIAN_VISIBILITY_DEFAULT BM25Weight : public Weight {
      *		   document length normalisation of wdf is.  0 means no
      *		   normalisation; 1 means full normalisation.  (default 0.5)
      *
+     *  @param delta  A parameter for pseudo tf value to control the scale
+     * 		   of the tf lower bound. Delta(δ) can be tuned for e.g from 0.0 
+     * 		   to 1.5 but BM25+ can still work effectively across collections 
+     *		   with a fixed δ = 1.0. (default 1.0)
+     * 
      *  @param min_normlen  A parameter specifying a minimum value for
      *		   normalised document length.  Normalised document length
      *		   values less than this will be clamped to this value, helping
      *		   to prevent very short documents getting large weights.
      *		   (default 0.5)
      */
-    BM25Weight(double k1, double k2, double k3, double b, double min_normlen)
+    BM25Weight(double k1, double k2, double k3, double b, double delta, double min_normlen)
```

Comment:
I think we should put the new parameter last; it'll make it easier for people to start using `delta` if they're putting a new number on the end rather than having to insert it in the middle of the list.

We also could do with an overloaded constructor that only has the old parameters, with `param_delta` being initialised to `1.0` as with the zero parameter constructor (so we don't break API compatibility with earlier releases).